src/pages/en/guides/data-fetching.md

```diff
@@ -4,51 +4,125 @@ title: Data Fetching
 description: Learn how to fetch remote data with Astro using the fetch API.
 ---
 
-Astro components and pages can fetch remote data to help generate your pages. Astro provides two different tools to pages to help you do this: **fetch()** and **top-level await.**
+`.astro` files can fetch remote data at build time to help generate your pages.
 
-## `fetch()`
+## `fetch()` in Astro
 
-Astro pages have access to the global `fetch()` function in their setup script. `fetch()` is a native JavaScript API ([MDN<span class="sr-only">- fetch</span>](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)) that lets you make HTTP requests for things like APIs and resources.
+All [Astro components](/en/core-concepts/astro-components) have access to the [global `fetch()` function](https://developer.mozilla.org/en-US/docs/Web/API/fetch) in their component script to make HTTP requests to APIs. This fetch call will be executed at build time, and the data will be available to the component template for generating dynamic HTML. 
 
-Even though Astro component scripts run inside of Node.js (and not in the browser) Astro provides this native API so that you can fetch data at page build time.
+💡 [**Top-level await**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await) is supported in your Astro component script.
```

Comment:
This line feels a bit out of place. Since the example below uses it, I don't think you _need_ to call it out here.

Or, if you want to keep a mention, then rephrase to make it clear the reason why you're bringing it up to the user. Something like "top-level await inside of your front matter script makes fetch() even easier to work with" or something like that.